### ISSUE TITLE
ELECTRA: Pre-training Text Encoders as Discriminators Rather Than Generators

### ISSUE THREAD

**Issue by seopbo** [ModuNLP/weekly-meeting#45]
## 어떤 내용의 논문인가요? 👋
- 본 논문의 기존의 pretrained language model 방법론들이 pretraining 시 많은 computing resource를 요구하는 점을 개선하기위해, sample-efficient한 pretraining 방법론인 replaced token detection을 제안하는 논문입니다.
- masked language model (BERT 계열), permutation language model (XLNET) 과는 다르게, transformer encoder로 각각 generator와 discriminator를 구성합니다.
- generator는 기존의 BERT 계열과 비슷하게 masked input에 대하여 masked language model의 loss로 backprop을 한 뒤 (15% masking), masked token에 대하여 token을 sample해서 corrupted input을 구성합니다. (eg. 논문에서는 여러가지 training 방법을 제안하고 있지만, 가장 default한 것을 적었습니다.)
- discriminator는 corrupted input의 모든 token에 대하여 각각의 token들이 sample된 것 인지, 원래 input에 있던 것인지를 구별하는 loss (real, fake prediction)를 통해 학습합니다
  - masked language model이 input의 subset tokens of input들에 대해서 loss를 계산하는 반면, 이 방법은 all tokens of input에 대해서 loss를 계산하므로 sample-efficient함
  - 또한 discriminator 입장에서 pretraining과 finetuning시에 discrepancy가 없음
  - 해당 loss는 generator에 backprop되지 않습니다.  이 때 sampled token이 원래 input의 token과 똑같다면, real로 간주합니다.
- pretraining이 완료되면 generator는 버리고, discriminator만 finetuning해서 사용합니다.
## Abstract (요약) 🕵🏻‍♂️
Masked language modeling (MLM) pre-training methods such as BERT corrupt the input by replacing some tokens with [MASK] and then train a model to reconstruct the original tokens. While they produce good results when transferred to downstream NLP tasks, they generally require large amounts of compute to be effective. As an alternative, ***we propose a more sample-efficient pre-training task called replaced token detection. Instead of masking the input, our approach corrupts it by replacing some tokens with plausible alternatives sampled from a small generator network.*** Then, instead of training a model that predicts the original identities of the corrupted tokens, ***we train a discriminative model that predicts whether each token in the corrupted input was replaced by a generator sample or not.*** Thorough experiments demonstrate ***this new pre-training task is more efficient than MLM because the task is defined over all input tokens rather than just the small subset that was masked out.*** As a result, ***the contextual representations learned by our approach substantially outperform the ones learned by BERT given the same model size, data, and compute.*** The gains are particularly strong for small models; for example, we train a model on one GPU for 4 days that outperforms GPT (trained using 30x more compute) on the GLUE natural language understanding benchmark. ***Our approach also works well at scale, where it performs comparably to RoBERTa and XLNet while using less than 1/4 of their compute and outperforms them when using the same amount of compute.***
## 이 논문을 읽어서 무엇을 배울 수 있는지 알려주세요! 🤔
- Contrastive Learning에 대한 일반론을 배울 수 있습니다.
## 같이 읽어보면 좋을 만한 글이나 이슈가 있을까요?
- [BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding](https://arxiv.org/abs/1810.04805)
- [XLNet: Generalized Autoregressive Pretraining for Language Understanding](https://arxiv.org/abs/1906.08237)
## 레퍼런스의 URL을 알려주세요! 🔗
https://openreview.net/forum?id=r1xMH1BtvB&utm_campaign=piqcy&utm_medium=email&utm_source=Revue%20newsletter
## code URL을 알려주세요! 🔗
https://github.com/ModuNLP/hacking_transformers/blob/master/20_07_02_%EB%B0%95%EC%9E%A5%EC%9B%90%EB%8B%98_ELECTRA.ipynb

**Comment by bage79** [comment]
[질문]

- 또한 discriminator 입장에서 pretraining과 finetuning시에 discrepancy가 없음
이 부분이 잘 이해되지 않는데요.
"discrepancy가 있는" 것은 어떤 경우일까요?
